clients/src/main/java/org/apache/kafka/common/telemetry/internals/SinglePointMetric.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.metrics.v1.AggregationTemporality;
+import io.opentelemetry.proto.metrics.v1.Metric;
+import io.opentelemetry.proto.metrics.v1.NumberDataPoint;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class represents a telemetry metric that does not yet contain resource tags.
+ * These additional resource tags will be added before emitting metrics by the telemetry reporter.
+ */
+public class SinglePointMetric implements MetricKeyable {
+
+    private final MetricKey key;
+    private final Metric.Builder metricBuilder;
+
+    private SinglePointMetric(MetricKey key, Metric.Builder metricBuilder) {
+        this.key = key;
+        this.metricBuilder = metricBuilder;
+    }
+
+    @Override
+    public MetricKey key() {
+        return key;
+    }
+
+    public Metric.Builder builder() {
+        return metricBuilder;
+    }
+
+    public static SinglePointMetric create(MetricKey metricKey, Metric.Builder metric) {
```

Comment:
Why do we add a `create` -- seems there is no validation happening (ie no additional code) -- seems easier to just make the constructor public?

Reply:
Yeah though it make sense, but I would let @xvrl to decide as these classes will be used elsewhere as well and @xvrl was the original author of this class.

Reply:
that's fine we can make the constructor public, it might just be a remnant of some refactoring.

Reply:
Moved to constructor, done.